src/lib/state/reducers/query.js

```diff
@@ -17,12 +17,24 @@ import {
   SET_QUERY_PAGINATION_SIZE,
   SET_QUERY_FILTERS,
   SET_QUERY_SUGGESTIONS,
+  SET_QUERY_STATE,
   SET_SUGGESTION_STRING,
   CLEAR_QUERY_SUGGESTIONS,
   RESULTS_UPDATE_LAYOUT,
   RESET_QUERY,
 } from '../types';
-import { updateQueryFilters } from '../selectors';
+import { updateQueryFilters, updateQueryState } from '../selectors';
+
+const store_keys = [
```

Comment:
we are multiplying this list of keys in multiple places, maybe better to define them in a file as const and import?